platform/solr/solr-factory-impl/src/main/java/org/codice/solr/factory/impl/SolrCloudClientFactory.java

```diff
@@ -275,4 +575,48 @@ private boolean isCollectionReady(CloudSolrClient client, String collection) {
       return false;
     }
   }
+
+  private boolean isAliasReady(CloudSolrClient client, String alias) {
+    try {
+      boolean aliasReady =
+          Failsafe.with(withRetry().retryWhen(false))
+              .onFailure(
+                  failure -> LOGGER.debug("Unable to get status on alias: {}", alias, failure))
+              .get(() -> aliasExists(alias, client));
+      if (!aliasReady) {
+        LOGGER.debug("Solr ({}): Timeout while waiting for Alias to exist", alias);
+      }
+      return aliasReady;
+    } catch (FailsafeException e) {
+      LOGGER.debug("Solr({}): Failure waiting for Alias to exist", alias, e);
+    }
+    return false;
+  }
+
+  protected void checkConfig() {
+    if (StringUtils.isBlank(zookeeperHosts)) {
+      zookeeperHosts = System.getProperty("solr.cloud.zookeeper");
+    }
+
+    if (StringUtils.isBlank(zookeeperHosts)) {
+      LOGGER.warn(
+          "Cannot create Solr Cloud client without Zookeeper host list system property [solr.cloud.zookeeper] being set.");
+      throw new IllegalStateException("system property 'solr.cloud.zookeeper' is not configured");
+    }
+  }
+
+  protected void storeConfigurationFiles(
+      File baseDir, List<SolrConfigurationData> configurationData) {
+    for (SolrConfigurationData configData : configurationData) {
+      File currentFile = new File(baseDir, configData.getFileName());
+      if (!currentFile.exists())
+        try (InputStream inputStream = configData.getConfigurationData();
```

Comment:
❓ This InputStream was opened in a much different class and is relying on this code to close it.  As a general rule, I don't like that design.  This call should be the one that opens (and closes) the InputStream and the configData object holds the information required to know what stream to open.  What happens if this function is never called?  Not sure if I followed it through well enough but does this get called if the collection already exists and is not getting recreated?  I don't think it does and so the configuration files would have a lingering open InputStream in that case.  There is a lot here so perhaps I missed that the InputStream can only get opened if you are for sure going to store the configuration files?  Even if that is the case, I would suggest changing the design to only open and close the stream here instead

Reply:
i am bother as well. but since each CollectionProvider have its own config file. The SolrCloudClientFactory doesn't have access to the config files. So the stream have to be passed in. The logic to open and close the stream are closely couple here. https://github.com/lamhuy/ddf/blob/splitcatalog/catalog/core/catalog-core-solr/src/main/java/ddf/catalog/source/solr/SolrSplitCatalogProvider.java#L490-L491
I did see a gap in the stream closing logic that will be added shortly